lib/collection/src/operations/snapshot_s3_ops.rs

```diff
@@ -0,0 +1,239 @@
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+
+use aws_sdk_s3::operation::complete_multipart_upload::CompleteMultipartUploadOutput;
+use aws_sdk_s3::operation::create_multipart_upload::CreateMultipartUploadOutput;
+use aws_sdk_s3::types::{CompletedMultipartUpload, CompletedPart};
+use aws_smithy_types::byte_stream::{ByteStream, Length};
+use aws_smithy_types_convert::date_time::DateTimeExt;
+
+use super::snapshot_ops::SnapshotDescription;
+use super::types::{CollectionError, CollectionResult};
+
+pub fn get_key(path: &Path) -> CollectionResult<String> {
+    // Get file name by trimming the path.
+    // if the path is ./path/to/file.txt, the key should be path/to/file.txt
+    let key = path.to_str().ok_or_else(|| {
+        CollectionError::s3_error(format!(
+            "Failed to get key for snapshot: {}",
+            path.display()
+        ))
+    });
+    key.map(|k| k.trim_start_matches("./").to_string())
+}
+
+pub async fn multi_part_upload(
```

Comment:
I used this official example code as the reference.

https://github.com/awsdocs/aws-doc-sdk-examples/blob/429c855e5f8d74330f9cbc4cb1570668439283ee/rustv1/examples/s3/src/bin/s3-multipart-upload.rs